File: src/py_pkg/py_pkg/subscriber.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SubscriberNode(Node):
    def __init__(self):
        super().__init__('subscriber')
        self.subscriber_ = self.create_subscription(String, 'topic', self.topic_callback, 10)
    
    def topic_callback(self, msg):
        self.get_logger().info("I heard the message : " + str(msg.data))

def main(args=None):
    rclpy.init(args=args)
    node = SubscriberNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: src/cpp_pkg/src/subscriber.cpp
#include "../include/cpp_pkg/subscriber.hpp"

Subscriber::Subscriber() : Node("subscriber")
{
    this->subscription_ = this->create_subscription<std_msgs::msg::String>("topic", 10, std::bind(&Subscriber::topic_callback, this, _1));
}

void Subscriber::topic_callback(const std_msgs::msg::String &msg) const
{
    RCLCPP_INFO(this->get_logger(), "%s", msg.data.c_str());
}

int main(int argc, char* argv[])
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<Subscriber>());
    rclcpp::shutdown();

    return 0;
}
